src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -386,6 +403,76 @@ private static String InternalReadAllText(String path, Encoding encoding)
                 return sr.ReadToEnd();
         }
 
+
+        // If we use the path-taking constructors we will not have FileOptions.Asynchronous set and
+        // we will have asynchronous file access faked by the thread pool. We want the real thing.
+        private static StreamReader AsyncStreamReader(string path, Encoding encoding)
+        {
+            FileStream stream = new FileStream(
```

Comment:
AFAICT. they aren't available to the corefx version of this, at least not when doing MSBuild from src/ or tests/